test/extensions/filters/network/ext_authz/ext_authz_test.cc

```diff
@@ -43,6 +43,8 @@ class ExtAuthzFilterTest : public testing::Test {
     client_ = new Filters::Common::ExtAuthz::MockClient();
     filter_ = std::make_unique<Filter>(config_, Filters::Common::ExtAuthz::ClientPtr{client_});
     filter_->initializeReadFilterCallbacks(filter_callbacks_);
+    time_system_.setMonotonicTime(std::chrono::milliseconds(0));
+    filter_->setStartTime(time_system_.monotonicTime());
```

Comment:
Instead, I think we can do this (before calling initializeReadFilterCallbacks):

```
auto time_system = std::make_unique<Event::SimulatedTimeSystem>();
time_system->setMonotonicTime(std::chrono::milliseconds(0));
time_system_ = time_system.get();
filter_callbacks_->connection_.dispatcher_.time_system_ = time_system;
```
(The `time_system_` field will need to be made into a pointer type.)

That makes the time system used in all these tests the simulated time you want and keeps a pointer to the time system available in the tests (just be careful not to use it when its invalid).

Then you shouldn't need to expose `setStartTime` or `getStartTime` in your filters.

Reply:
Did some digging into the MockDispatcher and found a GlobalTimeSystem that we can use instead.